build-system/tasks/dist.js

```diff
@@ -51,16 +51,52 @@ const {compileCss, cssEntryPoints} = require('./css');
 const {createCtrlcHandler, exitCtrlcHandler} = require('../ctrlcHandler');
 const {isTravisBuild} = require('../travis');
 const {maybeUpdatePackages} = require('./update-packages');
+const tempy = require('tempy');
+const conf = require('../build.conf');
+const {isCommonJsModule} = require('../compile/compile-utils');
 
 const {green, cyan} = colors;
 const argv = require('minimist')(process.argv.slice(2));
 
+
+const deglob = require('globs-to-files');
+const babel = require("@babel/core");
+
+const allJsSrcsGlob = [
+  '3p/**/*.js',
+  'ads/**/*.js',
+  'extensions/**/*.js',
+  'src/**/*.js',
+];
+
+function transferSrcsToTempDir() {
+  const tmp = `${tempy.directory()}/amphtml`;
+  process.env.AMP_TMP_DIR = tmp;
+  console.log('new tmp dir', tmp);
+  const files = deglob.sync(allJsSrcsGlob);
+  console.log('Executing babel transforms');
+  files.forEach(file => {
+    const {code} = babel.transformFileSync(file, {
+      plugins: conf.plugins({
+        isEsmBuild: argv.esm,
+        isCommonJsModule: isCommonJsModule(file),
```

Comment:
Can you check if this is still necessary? I added it during a previous closure compiler upgrade as a workaround to some errors I was seeing.

The call site is here:

https://github.com/ampproject/amphtml/blob/c4a663d0ba13d0488c6fe73c55dc8c971ac6ec0d/build-system/build.conf.js#L51-L55

Reply:
In other words, it would be nice to revert e4ea57103030da7e7959dba3eded96855e47af03 if possible.

Reply:
This is being removed in #23106

Reply:
ill remove once #23106 is merged on my end

Reply:
Merged.